packages/opentelemetry-basic-tracer/test/export/BatchSampledSpanProcessor.test.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import { BatchSampledSpanProcessor } from '../../src/export/BatchSampledSpanProcessor';
+import { NoopSpanExporter } from '../../src/export/NoopSpanExporter';
+import { NoopTracer } from '@opentelemetry/core';
+import { Span } from '../../src';
+
+const exporter = new NoopSpanExporter();
+const DEFAULT_BUFFER_SIZE = 10;
+const DEFAULT_BUFFER_TIMEOUT = 2000; // time in milliseconds
+const defaultBufferConfig = {
+  bufferSize: DEFAULT_BUFFER_SIZE,
+  bufferTimeout: DEFAULT_BUFFER_TIMEOUT,
+  logger: console,
+};
+
+describe('BatchSampledSpanProcessor', () => {
+  const tracer = new NoopTracer();
+
+  describe('constructor', () => {
+    it('should create a BatchSampledSpanProcessor instance', () => {
+      const processor = new BatchSampledSpanProcessor(
+        [exporter],
+        defaultBufferConfig
+      );
+      assert.ok(processor instanceof BatchSampledSpanProcessor);
+    });
+  });
+
+  describe('.onStart/.onEnd', () => {
+    it('should handle span started and ended', () => {
+      const processor = new BatchSampledSpanProcessor(
+        [exporter],
+        defaultBufferConfig
+      );
+      const span = new Span(tracer, 'span-name', {}, processor);
+      assert.strictEqual(processor['_spansDataList'].length, 0);
+      span.end();
+      assert.strictEqual(processor['_spansDataList'].length, 1);
+    });
+
+    it('should force flush when bufferSize exceeded', () => {
+      const processor = new BatchSampledSpanProcessor([exporter], {
+        bufferSize: 10,
+      });
+      for (let j = 0; j < 10; j++) {
+        processor.onEnd(new Span(tracer, 'span-' + j, {}, processor));
+      }
+      assert.strictEqual(processor['_spansDataList'].length, 10);
+      processor.onEnd(new Span(tracer, 'overflow-span', {}, processor));
+      assert.strictEqual(processor['_spansDataList'].length, 0);
+    });
+
+    // it('should force flush when timeout exceeded', done => {
```

Comment:
Why is this commented out? If it doesn't work yet, would you be open to just having a @todo comment and create an issue for it? (I'm a bit wary of checking in commented out code since it might not be clear to future people why it's there)